include/simdjson/generic/ondemand/array-inl.h

```diff
@@ -50,6 +50,11 @@ simdjson_really_inline simdjson_result<array> array::start(value_iterator &iter)
   SIMDJSON_TRY( iter.start_array().get(has_value) );
   return array(iter);
 }
+simdjson_really_inline simdjson_result<array> array::start_root(value_iterator &iter) noexcept {
+  simdjson_unused bool has_value;
```

Comment:
You do not have to do it, but a comment here explaining this construction might make the code easier to read.